torch/_jit_internal.py

```diff
@@ -707,11 +712,11 @@ def unused_method(self, x):
         # any compiled methods and wasn't decorated with `@torch.jit.export`
         m = torch.jit.script(MyModule())
     """
-    fn._torchscript_modifier = FunctionModifiers.EXPORT
+    fn._torchscript_modifier = FunctionModifiers.EXPORT  # type:ignore[attr-defined]
```

Comment:
All right, I see how this movie is going to play out and unfortunately, I think we're going to see this in a lot of the codebase as we move forward, because attaching an attribute to some function that didn't have it before is pretty endemic in this codebase. 😁 

And in the context of what we're doing, it's the right thing to do. My belief is that almost all the time we do this, it's because we've just wrapped a function or something that creates a brand new function and we need to give it "metadata".

So how to fix this problem, with an eye to the general case?

What about a `Protocol`? Is there a way to have a protocol that's callable with just one member, and a `TypeGuard`?

That was the best I came up with unfortunately.

Reply:
Something like:
```
class WithMeta(Protocol[P, R]):
    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> R: ...
    __meta__: dict[str, Any]  # or whatever shape your metadata has
```

Reply:
@rec Feel free to this cleanup in a follow up PR, this is unblocks a lot of type hints in the codebase so I'd like to merge this first before we optimize it to use Protocols

Reply:
Absolutely. I have a tendency to try to put too much into one pot when cooking. 😁 

Thanks for doing this, and for changing the message!